### ISSUE TITLE
horizontal rule render issue

### ISSUE THREAD

**Issue by whistyun** [whistyun/Markdown.Avalonia#24]
```md
Dashes:

---

    ---

    - - -

Asterisks:

***

    ***

* * *

    * * *

Underscores:

___

    ___

    _ _ _

```

output  
![image](https://user-images.githubusercontent.com/13712028/102704767-d5069880-42c2-11eb-981d-38c17236e32e.png)

**Comment by whistyun** [comment]
It is not problem of Horizontal rule but problem of list or codeblocks.
When we insert 4-spaces to linehead, it should be treat as [codeblock](https://www.markdownguide.org/basic-syntax#code-blocks).

**Comment by whistyun** [comment]
I should make test case more detailed. 
* for indent code block 
* for list indent boundaly condition

**Comment by whistyun** [comment]
this issue fixed in 0.10.0-a8 and 0.09.0-a8.

**Comment by whistyun** [comment]
this issue fixed in 0.10.0-a8 and 0.09.0-a8.	
![image](https://user-images.githubusercontent.com/13712028/103149549-d7c72900-47ad-11eb-814e-caade00b2a0b.png)
